php/WP_Mock/Hook.php

```diff
@@ -62,6 +65,14 @@ protected function safe_offset($value): string
         }
 
         if (is_object($value)){
+            if (! $value instanceof Type) {
+                $class = get_class($value);
+
+                if (isset(static::$objects[ $class ])) {
+                    return static::$objects[ $class ];
+                }
```

Comment:
```suggestion
                if (isset(static::$objects[$class]) && is_string(static::$objects[$class])) {
                    return static::$objects[$class];
                }
```
Adding a data type sanity check here would prevent the risk of fatal error in case the array holds a non-string value for some reason, considering this is a public static property.

Reply:
@nmolham-godaddy Good catch 👍. Thanks.